model/fn-execution/src/main/proto/beam_fn_api.proto

```diff
@@ -277,6 +291,21 @@ message ProcessBundleProgressRequest {
   string instruction_id = 1;
 }
 
+// A request to provide full MonitoringInfo for a given bundle.
+//
+// An SDK can report metrics using an identifier that only contains the
+// associated payload. A runner who wants to receive the full metrics
+// information can request all the monitoring metadata via a
+// ProcessBundleProgressMetadataRequest providing a list of ids as necessary.
+//
+// The SDK is allowed to reuse the identifiers across multiple bundles as long
+// as the MonitoringInfo could be reconstructed fully by overwriting its
+// payload field with the bytes specified here.
+message ProcessBundleProgressMetadataRequest {
+  // A list of ids for which the full MonitoringInfo is requested for.
+  repeated string id = 1;
```

Comment:
Maybe call this `monitioring_info_id` to distinguish this from the many other ids (like instruction id, process bundle id, etc.)

Reply:
Done